### PR TITLE
Ignoring key prefixes

### PR DESCRIPTION
**Pull request by mlitwiniuk**
I would like to be able to specify certain translation keys, that should not be added to backend (lit). Somethinkg like available_locales which let us specify only locales, that will be taken under consideration.

### CODE REVIEW

File: lib/lit/i18n_backend.rb [hunk 1]
```diff
@@ -129,5 +131,12 @@ def valid_locale?(locale)
       locales = ::Rails.configuration.i18n.available_locales
       !locales || locales.map(&:to_s).include?(locale.to_s)
     end
+
+    def is_ignored_key(key_without_locale)
+      Lit.ignored_keys.each do |k|
```
Comment (Bonias) [inline lib/lit/i18n_backend.rb]: You could use `any?`

    Lit.ignored_keys.any?{ |k| key_without_locale.start_with?(k) }

File: lib/lit/i18n_backend.rb [hunk 2]
```diff
@@ -51,12 +51,14 @@ def lookup(locale, key, scope = [], options = {})
...
       key_with_locale = parts.join('.')
+      _, key_without_locale = ::Lit::Cache.split_key(key_with_locale)
 
       ## check in cache or in simple backend
       content = @cache[key_with_locale] || super
       return content if parts.size <= 1
 
-      unless @cache.has_key?(key_with_locale)
+      if !@cache.has_key?(key_with_locale) &&
+          !is_ignored_key(key_without_locale)
```
Comment (Bonias) [inline lib/lit/i18n_backend.rb]: Maybe extract it into something like

    def should_cache?(key_with_locale)
      return false if @cache.has_key?(key_with_locale)

      _, key_without_locale = ::Lit::Cache.split_key(key_with_locale)
      return false if is_ignored_key(key_without_locale)

      true
    end

? (there is no need to split keys on every `translate` call)

Comment (mlitwiniuk) [inline lib/lit/i18n_backend.rb]: :+1: great catch

**Comment by Bonias** [comment]
Looks good now

**Pull request merged**